base/base_tests/stl_helpers_test.cpp

```diff
@@ -45,4 +45,44 @@ UNIT_TEST(CompareBy_Method)
   for (size_t i = 0; i < v.size(); ++i)
     TEST_EQUAL(v[i].Get(), static_cast<int>(i), ());
 }
+
+UNIT_TEST(SortUnique)
+{
+  vector<int> v = {1, 2, 1, 4, 3, 5, 2, 7, 1};
+  my::SortUnique(v);
+  vector<int> const expected = {1, 2, 3, 4, 5, 7};
+  TEST_EQUAL(v, expected, ());
+}
+
+UNIT_TEST(SortUniqueCompPredTest1)
+{
+  using TValue = int;
+  using TPair = pair<TValue, int>;
+  vector<TPair> v =
+      {{1, 22}, {2, 33}, {1, 23}, {4, 54}, {3, 34}, {5, 23}, {2, 23}, {7, 32}, {1, 12}};
+
+  my::SortUnique<TPair>(v, my::CompareBy(&TPair::first),
+                        [](TPair const & p1, TPair const & p2) { return p1.first == p2.first; });
```

Comment:
FYI: I'm going to rename CompareBy by LessBy, and implement a new helper - EqualsBy. So, the call will look like:
```c++
my::SortUnique(v, my::LessBy(&TPair::first), my:EqualsBy(&TPair::second));
```